compiler/rustc_mir_transform/src/dest_prop.rs

```diff
@@ -332,10 +351,27 @@ impl<'a, 'tcx> MutVisitor<'tcx> for Merger<'a, 'tcx> {
                     _ => {}
                 }
             }
+            StatementKind::Deinit(place) => {
+                if self.should_remove_write_at(place.local, location) {
+                    statement.make_nop();
+                }
+            }
 
             _ => {}
         }
     }
+
+    fn visit_operand(&mut self, op: &mut Operand<'tcx>, location: Location) {
+        self.super_operand(op, location);
+        match op {
+            Operand::Move(place) => {
+                if self.should_remove_write_at(place.local, location) {
+                    *op = Operand::Copy(*place);
```

Comment:
This is only valid if `place` is sized as unsized values can't be copied. I don't think `should_remove_write_at` can return true for unsized places (maybe there is a case where an unsized place can be re-initialized after a move though?), but it may be a good idea to check anyway.

Reply:
Hm, this is a theoretical problem in future versions of dest prop I think. God, I really wish we could just treat unsized values as completely unsupported. I'll add a check.